lib/widget/camera_widget.dart

```diff
@@ -0,0 +1,438 @@
+import 'dart:io';
+import 'dart:typed_data';
+import 'package:flutter/foundation.dart' show kIsWeb;
+import 'package:camera/camera.dart';
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+
+class Camera extends StatefulWidget
+    with Invokable, HasController<MyCameraController, CameraState> {
+  static const type = 'Camera';
+
+  Camera({Key? key}) : super(key: key);
+
+  final MyCameraController _controller = MyCameraController();
+
+  @override
+  MyCameraController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => CameraState();
+
+  @override
+  Map<String, Function> getters() {
+    return {};
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {};
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'height': (value) => controller.height = Utils.optionalDouble(value),
+      'width': (value) => controller.width = Utils.optionalDouble(value)
+    };
+  }
+}
+
+class MyCameraController extends WidgetController {
+  double? height;
+  double? width;
+
+  bool isFrontCamera = false;
+
+  List<CameraDescription>? cameras;
+  CameraController? cameracontroller;
+
+  final ImagePicker imagePicker = ImagePicker();
+  List<XFile> imageFileList = [];
+  List<Uint8List> imagesFileWeb = [];
+
+  Future<void> cameraFunction() async {
+    cameras = await availableCameras();
+    notifyListeners();
+  }
+
+  void initCamera() {
+    cameraFunction().then((value) {
+      setCamera(0);
+    });
+  }
+
+  void setCamera(int i) {
+    cameracontroller = CameraController(cameras![i], ResolutionPreset.max);
+    cameracontroller!.initialize().then((_) {
+      notifyListeners();
+    });
+  }
+
+  void selectImage() async {
+    final List<XFile>? selectImage = await imagePicker.pickMultiImage();
+    if (selectImage!.isNotEmpty) {
```

Comment:
don't use !. The selectImage also cannot be null according to the function call.

Reply:
fixed